### PR TITLE
Alsatian/server refactor

### PR DESCRIPTION
**Pull request by tillywoodfield**

### CODE REVIEW

File: server/services/gameLogic.js [hunk 1]
```diff
@@ -1,34 +1,66 @@
...
+    cardId
+} */
 let votes = [];
 
-/* Return the list of players */
-exports.getAllPlayers = () => {
-    return allPlayers;
+/* Return the list of players, without their assigned cards */
+exports.getPlayers = () => {
+    return players.map(player => ({ username: player.username, score: player.score }));
```
Comment (Ro4052) [inline server/services/gameLogic.js]: Another one that can be a one-liner.

Comment (Ro4052) [reply server/services/gameLogic.js]: Same with other functions in this file.

**Review by sibela46 — DISMISSED** [review]

File: server/services/gameLogic.js [hunk 2]
```diff
@@ -51,91 +83,101 @@ exports.quitGame = player => {
...
+const playerHasVoted = (username) => {
+    return votes.filter(vote => vote.username === username).length > 0;
 }
 
 /* Returns true if all players have played a card this round */
 const allPlayersPlayedCard = () => {
-    for (let player of allPlayers.filter(player => currentPlayer !== player)) {
-        if (!player.playedCard) return false;
+    for (let player of players) {
+        if (!playerHasPlayedCard(player.username)) return false;
```
Comment (Ro4052) [inline server/services/gameLogic.js]: These two `for` loops in `allPlayersPlayedCard` and `allPlayersVoted` could be implemented using `every`.
`const allPlayersPlayedCard = () => players.every(player => playerHasPlayedCard(player))`
Something like that.

File: src/components/Cards/PlayedCards/PlayedCards.jsx [hunk 3]
```diff
@@ -4,9 +4,10 @@ import styles from '../Cards.module.css';
 
 export class PlayedCards extends React.Component {
     render() {
+        console.log(this.props.cards);
         return (
             <ul id="played-cards">
-                {this.props.cards.map(card => <img id={"card-" + card.id} className={styles.allCards} alt={"card-" + card.id} key={card.id} src={require(`../cards/card (${card.id}).jpg`)}/>)}
+                {this.props.cards.map(card => <img id={"card-" + card.cardId} key={card.cardId} className={styles.allCards} alt={"card-" + card.cardId} key={card.id} src={require(`../cards/card (${card.cardId}).jpg`)}/>)}
```
Comment (Ro4052) [inline src/components/Cards/PlayedCards/PlayedCards.jsx]: I could be wrong but I don't think you actually need to `require` the image file here. You might be able to just use relative paths.

File: src/components/PlayerInfo/Players.jsx [hunk 4]
```diff
@@ -8,7 +8,7 @@ export class Players extends React.PureComponent {
             <div className={style.currentPlayersBox}>
                 <h3>Players:</h3>
                 <ul id="players">
-                    {this.props.allPlayers.map((player, key) => {
+                    {this.props.players.map((player, key) => {
                         return <li key={key}>{player.username}</li>
```
Comment (Ro4052) [inline src/components/PlayerInfo/Players.jsx]: OK, I'll stop pointing out unecessary returns now, you get the point.

**Review by sibela46 — APPROVED** [review]

**Pull request merged**